tiledb/sm/query/reader.cc

```diff
@@ -640,32 +641,38 @@ Status Reader::compute_result_cell_slabs(
   STATS_FUNC_OUT(reader_compute_cell_ranges);
 }
 
-template <class T>
 Status Reader::compute_range_result_coords(
     unsigned frag_idx,
     ResultTile* tile,
-    const std::vector<const T*>& range,
+    const std::vector<const void*>& range,
     std::vector<ResultCoords>* result_coords) const {
-  auto dim_num = array_schema_->dim_num();
-  assert(dim_num == range.size());
-  const auto& t = tile->coords_tile();
-  // TODO: find coords_num in a different way
-  auto coords_num = t.cell_num();
-  auto c = (T*)t.internal_data();
-
-  for (uint64_t i = 0, pos = 0; i < coords_num; ++i, pos += dim_num) {
-    // TODO: must fix - it should be one coord per dimension
-    // TODO: does not need to be a separate function
-    // TODO: not templates here
-    if (utils::geometry::coords_in_rect<T>(&c[pos], range, dim_num) &&
-        !coords_overwritten(frag_idx, &c[pos]))
-      result_coords->emplace_back(tile, i);
+  auto coords_num = tile->cell_num();
+  auto fragment_num = fragment_metadata_.size();
+
+  for (uint64_t pos = 0; pos < coords_num; ++pos) {
+    // Check if the coordinates are in the range
+    if (!tile->coord_in_rect(pos, range))
+      continue;
+
+    // Check if the coordinates are overwritten by a future dense fragment
+    bool overwritten = false;
+    for (unsigned f = frag_idx + 1; !overwritten && f < fragment_num; ++f) {
+      if (fragment_metadata_[f]->dense()) {
```

Comment:
I don't see a condition on `dense()` in the previous code here, did this function previously check both sparse and dense?

Reply:
Nevermind, I see the condition was here instead: https://github.com/TileDB-Inc/TileDB/pull/1481/files#diff-e128e516e0718dc06066e0c7aecf7824L1989